Build a specific version of webpushd for the WK_RELOCATABLE_FRAMEWORKS build to an override directory

beidson opened the pull request:
#### fdabe492fd9a0ce66358c9b2821ef83f1a493011
<pre>
Build a specific version of webpushd for the WK_RELOCATABLE_FRAMEWORKS build to an override directory
<a href="https://bugs.webkit.org/show_bug.cgi?id=261010">https://bugs.webkit.org/show_bug.cgi?id=261010</a>
rdar://114801730

Reviewed by Elliott Williams and Alex Christensen.

This patch does two buckets of things for the WK_RELOCATABLE_FRAMEWORKS build to an override directory:
1 - WebPushDaemonMain listens on a different service name, connects to apsd with a different service name, and creates a seperate push database.
2 - webpushd has a different product name, install location, different sandbox profile, and different launchd plist

* Source/WebKit/Configurations/BaseTarget.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Configurations/webpushd.xcconfig:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/webpushd/WebPushDaemonMain.mm:
(WebKit::applySandbox):
(WebKit::WebPushDaemonMain):
* Source/WebKit/webpushd/com.apple.webkit.webpushd.relocatable.mac.plist: Added.
* Source/WebKit/webpushd/mac/com.apple.WebKit.webpushd.relocatable.mac.sb.in: Added.

Canonical link: <a href="https://commits.webkit.org/267636@main">https://commits.webkit.org/267636@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2e14ab2161674d92c3d178eabd0f79185ffa320c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17076 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17401 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17883 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18862 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15976 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20675 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17543 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18189 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17279 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17639 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14816 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19678 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14878 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22208 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15878 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15674 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20025 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16262 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13793 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15419 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4117 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19784 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16095 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/webpushd/WebPushDaemonMain.mm:
```diff
@@ -47,8 +47,15 @@
+static const ASCIILiteral defaultMachServiceName = "com.apple.webkit.webpushd.relocatable.service"_s;
+static const ASCIILiteral defaultIncomingPushServiceName = "com.apple.aps.webkit.webpushd.relocatable.incoming-push"_s;
+#else
+garbageline
```
remove

achristensen07 commented on Source/WebKit/webpushd/WebPushDaemonMain.mm:
```diff
@@ -47,8 +47,15 @@
 
 static const ASCIILiteral entitlementName = "com.apple.private.webkit.webpush"_s;
+
+#if defined(RELOCATABLE_WEBPUSHD)
```
I like #ifdef better, but this is fine.
Also, we might want to make this fit into the USE(RELOCATABLE_WEBPUSHD) convention we have elsewhere.

achristensen07 commented on Source/WebKit/webpushd/WebPushDaemonMain.mm:
```diff
@@ -143,7 +157,13 @@ int WebPushDaemonMain(int argc, char** argv)
             String libraryPath = NSSearchPathForDirectoriesInDomains(NSLibraryDirectory, NSUserDomainMask, YES)[0];
+
+#if defined(RELOCATABLE_WEBPUSHD)
+            String pushDatabasePath = FileSystem::pathByAppendingComponents(libraryPath, { "WebKit"_s, "WebPush"_s, "PushDatabase.relocatable.db"_s });
```
It would be slightly more concise to just have the ASCIILiteral have 2 definitions instead of duplicating the whole line.

achristensen07 commented on Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
```diff
@@ -424,6 +424,7 @@
 (allow mach-lookup (global-name "org.webkit.pcmtestdaemon.service"))
 
 (allow mach-lookup (global-name "com.apple.webkit.webpushd.service"))
+(allow mach-lookup (global-name "com.apple.webkit.webpushd.relocatable.service"))
```
We don't need both of these.  We will only ever need one or the other.

emw-apple commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -4885,6 +4887,7 @@
 		51240EB9220A08CA005CFC63 /* DownloadMap.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; path = DownloadMap.cpp; sourceTree = "<group>"; };
 		51240EBE220B6947005CFC63 /* NetworkResourceLoadMap.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = NetworkResourceLoadMap.h; sourceTree = "<group>"; };
 		51240EBF220B6948005CFC63 /* NetworkResourceLoadMap.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; path = NetworkResourceLoadMap.cpp; sourceTree = "<group>"; };
+		5124D5272AA19DCB005229E1 /* com.apple.WebKit.webpushd.relocatable.mac.sb */ = {isa = PBXFileReference; lastKnownFileType = text; name = com.apple.WebKit.webpushd.relocatable.mac.sb; path = ../../../../../../Users/bradeeoh/build/Release/DerivedSources/WebKit/com.apple.WebKit.webpushd.relocatable.mac.sb; sourceTree = "<group>"; };
```
Bad path! I think this needs to be:

```suggestion
		5124D5272AA19DCB005229E1 /* com.apple.WebKit.webpushd.relocatable.mac.sb */ = {isa = PBXFileReference; lastKnownFileType = text; name = com.apple.WebKit.webpushd.relocatable.mac.sb; path = DerivedSources/WebKit/com.apple.WebKit.webpushd.relocatable.mac.sb; sourceTree = BUILT_PRODUCTS_DIR; };
```

emw-apple reviewed (changes requested):
Marking r- to make sure you get signoff from me on the build system bits.

One of the other things to note is that WK_RELOCATABLE_FRAMEWORKS includes Debug/Release ("at-desk") builds. Naively, it seems like we would want Debug/Release webpushd to behave like the STP/relocatable variant, since it has to do similar things like coexist with the webpushd provided by the system.

beidson committed:
Build a specific version of webpushd for the WK_RELOCATABLE_FRAMEWORKS build to an override directory
https://bugs.webkit.org/show_bug.cgi?id=261010
rdar://114801730

Reviewed by Elliott Williams and Alex Christensen.

This patch does two buckets of things for the WK_RELOCATABLE_FRAMEWORKS build to an override directory:
1 - WebPushDaemonMain listens on a different service name, connects to apsd with a different service name, and creates a seperate push database.
2 - webpushd has a different product name, install location, different sandbox profile, and different launchd plist

* Source/WebKit/Configurations/BaseTarget.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Configurations/webpushd.xcconfig:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/webpushd/WebPushDaemonMain.mm:
(WebKit::applySandbox):
(WebKit::WebPushDaemonMain):
* Source/WebKit/webpushd/com.apple.webkit.webpushd.relocatable.mac.plist: Added.
* Source/WebKit/webpushd/mac/com.apple.WebKit.webpushd.relocatable.mac.sb.in: Added.

Canonical link: https://commits.webkit.org/267636@main
